### PR TITLE
Fix zero exit code when init failures during test execution

### LINKED ISSUE #41815: [Bug]: bal test command returning exit-code 0 on failure
### Description

The `bal test` command exits with an exit code of 0 when encountering a module-level error.

### Steps to Reproduce

For instance, if there is a global variable in the test code, like the example below:

```ballerina
final Client gmailClient = check new (gmailConfig);
```

If an error occurs during the execution of this code, the `bal test` command exits with code 0.  It's expected to return exit code 1.

### Affected Version(s)

Tested on: 2201.8.4

### OS, DB, other environment details and versions

_No response_

### Related area

-> Test Framework

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #41815

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java [hunk 1]
```diff
@@ -455,6 +458,8 @@ private void handleErrorFromFutureValue(MethodVisitor mv, boolean isTestFunction
         if (isTestFunction) {
             mv.visitFieldInsn(GETFIELD , FUTURE_VALUE , PANIC_FIELD, GET_THROWABLE);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java:459]: mv.visitFieldInsn(GETFIELD , FUTURE_VALUE , PANIC_FIELD, GET_THROWABLE) this is common to both if and else

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SingleTestExecutionWithInitFailuresTest.java
Comment (Nadeeshan96) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SingleTestExecutionWithInitFailuresTest.java:1]: Shall we add a test having a multi module project like this: default module imports module A and D, module A imports B, module B imports module C. We only have only a module init failure at module B. Some test like this.

Comment (gabilang) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/SingleTestExecutionWithInitFailuresTest.java:1]: Added the same scenario

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**